reflex/state.py

```diff
@@ -2425,6 +2426,200 @@ async def modify_state(self, token: str) -> AsyncIterator[BaseState]:
             await self.set_state(token, state)
 
 
+def _default_token_expiration() -> int:
+    """Get the default token expiration time.
+
+    Returns:
+        The default token expiration time.
+    """
+    return get_config().redis_token_expiration
+
+
+def _purge_expired_states(token_expiration: int):
+    """Purge expired states from the disk.
+
+    Args:
+        token_expiration: The token expiration time in seconds.
+    """
+    import time
+
+    for path in path_ops.ls(prerequisites.get_web_dir() / constants.Dirs.STATES):
+        # check path is a pickle file
+        if path.suffix != ".pkl":
+            continue
+
+        # load last edited field from file
+        last_edited = path.stat().st_mtime
+
+        # check if the file is older than the token expiration time
+        if time.time() - last_edited > token_expiration:
+            # remove the file
+            path.unlink()
+
+
+class StateManagerDisk(StateManager):
+    """A state manager that stores states in memory."""
+
+    # The mapping of client ids to states.
+    states: Dict[str, BaseState] = {}
+
+    # The mutex ensures the dict of mutexes is updated exclusively
+    _state_manager_lock = asyncio.Lock()
+
+    # The dict of mutexes for each client
+    _states_locks: Dict[str, asyncio.Lock] = pydantic.PrivateAttr({})
+
+    # The token expiration time (s).
+    token_expiration: int = pydantic.Field(default_factory=_default_token_expiration)
+
+    class Config:
+        """The Pydantic config."""
+
+        fields = {
+            "_states_locks": {"exclude": True},
+        }
+
+    def __init__(self, state: Type[BaseState]):
+        """Create a new state manager.
+
+        Args:
+            state: The state class to use.
+        """
+        super().__init__(state=state)
+
+        states_directory = prerequisites.get_web_dir() / constants.Dirs.STATES
+        path_ops.mkdir(states_directory)
+
+        _purge_expired_states(self.token_expiration)
```

Comment:
i was kind of expecting `states_directory` to become a (cached) property, and `_purge_expired_states` to be a method, so it doesn't need to also determine the directory or be passed the token expiration